drivers/mrf24j40/mrf24j40_netdev.c

```diff
@@ -57,6 +83,11 @@ static int _init(netdev_t *netdev)
     gpio_set(dev->params.reset_pin);
     gpio_init_int(dev->params.int_pin, GPIO_IN, GPIO_RISING, _irq_handler, dev);
 
+    /* before proceeding, check if the device responds */
+    if (_test_spi_connection(dev)) {
```

Comment:
Any particular reason to move the check to the netdev implementation of the driver? I could still be performed in `mrf24j40_init` but before the reset right?

Reply:
`mrf24j40_init()` gets called by `mrf24j40_reset()`  but that only gets called in two paces in `mrf24j40_netdev.c` - so yea, I can move it back.

Reply:
I moved it back and updated the `mrf24j40_reset()` and `mrf24j40_init()` functions to return an error code.